components/script/dom/radionodelist.rs

```diff
@@ -0,0 +1,165 @@
+                                // Step 3-4
+                                let value = input.Value();
+                                Some(if value.is_empty() { DOMString::from("on") } else { value })
+                            }
+                            _ => None
+                        }
+                    })
+                }).next()
+                // Step 2
+                  .unwrap_or(DOMString::from(""))
+            }
+        }
+    }
+
+    // https://html.spec.whatwg.org/multipage/infrastructure.html#dom-radionodelist-value
+    fn SetValue(&self, value: DOMString) {
+        match *self.upcast::<NodeList>().get_list_type() {
+            NodeListType::Simple(ref v) => {
+                for node in v.iter() {
+                    // Step 1
+                    if let Some(input) = node.downcast::<HTMLInputElement>() {
+                        match input.type_() {
+                            atom!("radio") if value == DOMString::from("on") => {
+                                // Step 2
+                                let val = input.Value();
+                                if val.is_empty() || val == DOMString::from("on") {
+                                    input.SetChecked(true);
+                                    return;
+                                }
+                            }
+                            atom!("radio") => {
+                                // Step 2
+                                if input.Value() == value {
+                                    input.SetChecked(true);
+                                    return;
+                                }
+                            }
+                            _ => {}
+                        }
+                    }
+                }
+            }
+            NodeListType::Children(ref cl) => {
+                for node in cl.get_parent_node().traverse_preorder() {
+                    // Step 1
+                    if let Some(input) = node.downcast::<HTMLInputElement>() {
+                        match input.type_() {
+                            atom!("radio") if value == DOMString::from("on") => {
+                                // Step 2
+                                let val = input.Value();
+                                if val.is_empty() || val == DOMString::from("on") {
+                                    input.SetChecked(true);
+                                    return;
+                                }
+                            }
+                            atom!("radio") => {
+                                // Step 2
+                                if input.Value() == value {
+                                    input.SetChecked(true);
+                                    return;
+                                }
+                            }
+                            _ => {}
+                        }
+                    }
+                }
+            }
+        };
```

Comment:
Probably not. `while let Some(node) = cl.parent_node().traverse_preorder().next()` doesn't seem to help here since we'd still need to downcast the node.